### PR TITLE
fix: minor version parsing in version compatibility check

### PR DESCRIPTION
**Pull request by a7i**
fixes in #1419 

/cc @damemi

### CODE REVIEW

File: pkg/version/version_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import (
+	"testing"
+)
+
+func TestGet(t *testing.T) {
+	tests := []struct {
+		name      string
+		version   string
+		wantMajor string
+		wantMinor string
```
Comment (damemi) [inline pkg/version/version_test.go:12]: could you just have one version object and check that it's equal to the expected? The diff in the output of that check will show whether it's the major or minor that's wrong

Comment (a7i) [reply pkg/version/version_test.go:12]: a lot of the fields need to be set to do a reflectEqual, let me push a separate commit to get your feedback

Comment (a7i) [reply pkg/version/version_test.go]: let me know what you think

Comment (a7i) [reply pkg/version/version_test.go]: any feedback?

File: pkg/descheduler/descheduler.go [hunk 2]
```diff
@@ -276,30 +277,30 @@ func Run(ctx context.Context, rs *options.DeschedulerServer) error {
...
-	serverVersion, err := utilversion.ParseSemantic(serverVersionInfo.String())
+	kubeServerVersion, err := utilversion.ParseSemantic(kubeServerVersionInfo.String())
 	if err != nil {
 		return errors.New("failed to parse Kubernetes server version")
 	}
 
-	deschedulerVersion, err := utilversion.ParseGeneric(versionInfo.GitVersion)
+	deschedulerMinor, err := strconv.ParseFloat(deschedulerVersionInfo.Minor, 64)
 	if err != nil {
 		return errors.New("failed to convert Descheduler minor version to float")
```
Comment (ingvagabund) [inline pkg/descheduler/descheduler.go:293]: Could we also print `deschedulerVersionInfo.Minor` in the error message? So users can see the invalid version string.

File: pkg/descheduler/descheduler.go [hunk 3]
```diff
@@ -276,30 +277,30 @@ func Run(ctx context.Context, rs *options.DeschedulerServer) error {
...
+func validateVersionCompatibility(discovery discovery.DiscoveryInterface, deschedulerVersionInfo version.Info) error {
+	kubeServerVersionInfo, err := discovery.ServerVersion()
 	if err != nil {
 		return errors.New("failed to discover Kubernetes server version")
 	}
 
-	serverVersion, err := utilversion.ParseSemantic(serverVersionInfo.String())
+	kubeServerVersion, err := utilversion.ParseSemantic(kubeServerVersionInfo.String())
 	if err != nil {
 		return errors.New("failed to parse Kubernetes server version")
```
Comment (ingvagabund) [inline pkg/descheduler/descheduler.go:288]: Could we also print `kubeServerVersionInfo.String()` in the error message? So users can see the invalid version string.

**Pull request merged**